### ISSUE TITLE
[Bug] Don't save downloads to application directory

### ISSUE THREAD

**Issue by hXtreme** [MCMrARM/revolution-irc#265]
Frankly there's no good reason to save download files to application directory so the option is just a bloat and not only that but very disruptive as those files can't be accessed/processed/transferred easily.

**Comment by MCMrARM** [comment]
I am mixed about it being a bug. The reason it's there is so you don't have to give the app Downloads dir permission.

**Comment by hXtreme** [comment]
But anything saved there can't be accessed from outside the app. So maybe there should be a way to move data out of the application directory.

@MCMrARM I'm really sorry if my issues sound a bit harsh, back then I had spent a good few hours downloading some files on a limited data connection and when it turned out that I couldn't transfer it to my laptop I just broke down and became very frustrated and angry.

Thanks for working on this app, and sorry once again.

**Comment by MCMrARM** [comment]
I think it saves it in external storage by default so it should be saved to Android/data/io.mrarm.irc/files by default I think.

I mean I am pretty sure the app asks for the Download dir permission on the first download though.

**Comment by hXtreme** [comment]
I believe contents of Android/data aren't accessible without root permissions.

Yes the app does ask for permission but anything that's saved in the app directory isn't accessible.

Follow the below steps:

1. Set download directory to app directory.
2. Download a file.
3. Try to copy the file to a computer (This step wasn't possible for me)

**Comment by MCMrARM** [comment]
They are accessible using most file explorers just fine. Unless you have no Internal Storage but that's kind of rare.

Android MTP is broken as hell and as such I would heavily recommend against using it in any reproduction steps. Please try accessing the directory with a pre-Android 11 file manager.

**Comment by hXtreme** [comment]
Unfortunately I don't have any pre Android 11 devices.

The following is a screenshot of what I get when I try to access the data folder:

![Screenshot_20201124-094645](https://user-images.githubusercontent.com/22425371/100120274-c785fc00-2e45-11eb-942f-dffa0117e16c.png)


related: https://support.google.com/pixelphone/thread/70501031?hl=en&msgid=72762121

**Comment by MCMrARM** [comment]
Ah so it changed with Android 11. This is more of a request to properly support Android 11 then. I haven't looked into what changed in Android 11 yet, but it seems like some new API will have to be used to expose the directory.

**Comment by hXtreme** [comment]
That or you could add an option to move a downloaded file to a specific
directory.

Best,
Harsh

**Comment by MCMrARM** [comment]
If possible I'd like to keep an app-internal directory since I do not want to require read permission to Downloads folder if I don't need it. I think Android 10 and 11 introduced APIs which might actually deprecate the need to request the Downloads folder permission altogether, so preferably these should be used. I'll try to look into proper Android 11 support once I manage to find a stable Android 11 ROM for my OnePlus 5 (or get a new phone).

**Comment by hXtreme** [comment]
Frankly I was just hoping if you could add an additional option called either `move` or `export` right above the delete option that already exists for downloads.

And you can ask for permission if and only if someone wants to move it to a place where the app doesn't have permission to write to.

---

You should be able to use the Android emulator bundled with the IDE to run Android 11.

---

Also, I don't think I've heard anything about Android eventually not requiring permission for Download directory. That actually sounds opposite of what they've been doing recently.

**Comment by MCMrARM** [comment]
Nah, I am pretty sure they introduced a way to have per-app Download directories and then show it together in the Downloads app. I am not sure how it works since I only skimmed over it somewhere.

I do not like the Android emulator and would rather get a real device when possible.

**Comment by kurahaupo** [comment]
This would seem to be related to #46

**Comment by kurahaupo** [comment]
This would resolve #264

**Comment by Bakaba** [comment]
@MCMrARM In what world do you live? You have guts to try to justify something like this. scam app, uninstalled.

**Comment by hXtreme** [comment]
@Bakaba chill out, if you have such a big problem this is open source fork it and fix it yourself no need to be toxic.

**Comment by kurahaupo** [comment]
@MCMrARM I thought that Android only asks the user to grant permissions when the app first attempts an operation that requires that permission, in which case it shouldn't matter having it in the manifest.
